---
title: "R Notebook"
output: html_notebook
---

Department of Plant Science - Departmental Graduate Student Research Assistantships

Due date: March 15, 2017 (??)

Material to be submitted:
- Research proposal
- Resume or CV
- Letter of Reference
- Undergraduate and Graduate Transcripts
- GRE scores

Proposal:
  - Relevance or justification
  - Goals and objectives
  - Experimental Plan
  
My general thoughts:

- model the genotypic impact on measurable traits
  - meaning create a model to predict traits based on a model including genotype and kinship

Overall drive of the project will be to look for modules of genes (gene clusters) that have undergone
selection for certain traits.

1. Gene modules using WGCNA
2. use the eigenvalues from the modules (the PC1) as the trait and model using the genotype values and kinship as fixed effect
  2.a. do I need to add covariates? What are the covariates in this case?
3. use the FPKM of the each of the genes within a module to do GWAS on each of those genes
4. Collect the top SNPs from all of the gwas studies and perform a Qx to look for regions of the genome that have diverged (been under selection)

So introduction topics will be:
  - The need to understand how to predict traits
  - simple ways this has been implimented
  - Issues in GWAS - relateness, and how to deal with it
  - simplifying genetic complexity via creating modules
  - GWAS background
  - potential uses for this type of approach

Several calls for re-thinking the statistical approaches for calculating 

OR dealing with the small-effect/latent genes that also have a cumulative major impact on traits, but
go unmeasured because of statistical power/their small effects
  
Modeling can be useful for seeing outliers, looking for adaptation within measured groups, seeing adaptation 
outside of groups.
  
What results in traits
  - genomic quality (dominance, heritability, effect size)
  - selection (adaptation to environment)
  - drift
  - population structure
  
Goals and Objectives
  - Model
  - Incorporate 
  
==========

Current benefits of GWAS
GWA studies are used to find association/correlation between a trait and a genomic marker, often a signle
nucleotide polymorphism, and are used in almost all fields of biological research. However, each organism
and population, and research goal, introduces new assumptions and limitations to a GWA study.
Genome wide association (GWA) studies investigating association between genes and traits of interest
have led to the discovery and characterization of many causal genes in maize [citations].
Something about the limitations and assumptions of 


=========
Q_x is the way it will be

So, I'm finishing the GWAS on the traits

Then doing GWAS on the modules, using FPKM as the traits
  - here, eigengenes
  
Then, doing GWAS on the genes within each module, and looking for small cis effects (still using FPKM as the trait?)

This will produce a set of SNPs, some that are invluential to the module as a whole (from the eigengene GWAS) and 
others that are more important to an individual gene within the module (SNPs from individual gene GWAS).

From this set of SNPs, perform a Qx analysis, which will look for the ditribution of the traits among populations
that is not explained by relatedness. A positive hit in Qx indicates that that SNP has variation that is different 
(probably a higher degree of variance) than the expected MVN distribution.

Where this is tricky is... what does this mean? That the eigengene has a SNP that has been under selection, so all the genes
in the module are underselection?






