---
title: "Graphing gemma lmm output"
output:
  html_document:
    self_contained: no
  html_notebook: default
---

Ran GEMMA on traits and relatedness matrix

```{r}
setwd("~/Dropbox/UCD/rotations/RRI/maize_genomics_project")
```

Load libraries

```{r}
library(gplots)
library(qqman)
```

#Phenotype traits 

Trait names

```{r}
traits <- read.table("data/gemma_input/traits.txt", sep=",")
#get ride of the first value, which is the word "taxa"
traits <- traits[,-1]
traits
```

(Wisconsin Diversity Panel Phenotypes)  
GDD: days to flowering (growing degree days)  
kernel300 - weight of 300 kernels (g)  
stalk - stalk diameter (cm)  
plant_height - cm  
ear_height - cm  
leaf_number - number of leaves  
stover_yield - $t ha^{-1}$  
juvenile - last leaf with epicuticular wax (length? what is this?)  
ear_internode - uppermost internode with developed ear (unit?)  
internode_length - cm  
internode\_above_ear - number of internodes above uppermost ear  
adult - number of leaves with no epicuticular wax  
percent_juvenile - percent of leave with epicuticular wax  
percentage_adult - percent of leaves with no epicuticular wax  

#Data processing

Maize Genotypes were downloaded from panzea genotype data. I used the Maize GBS genotypes version 2.7 (http://cbsusrv04.tc.cornell.edu/users/panzea/download.aspx?filegroupid=4)

Specifically, the file ZeaGBSv27_publicSamples_imputedV5_AGPv4-161010.h5, imputed genotypes with ACPv4 coordinates.

To process this file, this was first converted into hapmap format, then converted into VCF [make one step?]. In addition, the panzea
data has genotype data for 17,280 public maize lines. The genotype data was subsetted with a list of maize lines used for the 
Wisconsin diversity panel.

Filtering of the genotypes was done with vcftools.

vcftools was used for minor allele frequency, set to 0.01.

Frequency of missing data at the filtered sites:

```{r}
lmiss <- read.table("data/lowerFreq/ZeaGBSv27_sf_vcfInfo.lmiss", sep="\t", header=T)

hist(lmiss$F_MISS)
max(lmiss$F_MISS)

#number of missing data with >20%
nrow(lmiss[lmiss$F_MISS > 0.2,])

imiss <- read.table("data/lowerFreq/ZeaGBSv27_sf_vcfInfo.imiss", sep="\t", header=T)
hist(imiss$F_MISS)
max(imiss$F_MISS)

#lines with above 40% missing
bad_lines <- imiss[imiss$F_MISS >= 0.4,]
nrow(bad_lines)


#Remove these lines from the list of taxa used to subset the hdf5 from tassel
taxa <- read.table("data/gemma_input/ZeaGBS_phenotypeLines_T.txt")
taxa <- as.character(taxa$V1)
length(taxa)
bad_lines <- as.character(bad_lines$INDV)
t <- taxa %in% bad_lines
length(t)
taxa <- taxa[!t]
length(taxa)
#write.table(taxa, "data/gemma_input/ZeaGBS_phenotypeLines_c_T.txt", sep=" ", quote = F, col.names = F, row.names = F)

#also for the taxa list for python
taxa <- read.table("data/gemma_input/ZeaGBS_phenotypeLines.txt")
taxa <- as.character(taxa$V1)
t <- taxa %in% bad_lines
sum(t)
taxa <- taxa[!t]
#write.table(taxa, file="data/gemma_input/ZeaGBS_phenotypeLines_c.txt",sep=" ", quote = F, col.names = F, row.names = F)
```

Using this modified list of maize lines, re-subset the original panzea data (or could just have removed from the already subsetted file, but I just
reran the pipeline.)

Filter with vcftools --maf 0.01:

```{r}
#imiss <- read.table("data/trait_gwas_output/ZeaGBSv27_s_vcfInfo.imiss", sep="\t", header=T)
#lmiss <- read.table("data/trait_gwas_output/ZeaGBSv27_s_vcfInfo.lmiss", sep="\t", header=T)
#hist(imiss$F_MISS, breaks = 20, main = "Frequency of Misisng Data in Individuals")
#hist(lmiss$F_MISS, breaks=20, main="Frequency of Missing Data in Loci")
#(max(lmiss$F_MISS))
```

###Site frequency spectrum

In progress


#Relatedness  matrix

Used GEMMA to construct a relatedness matrix

Relatedness - either centered or standardized From the GEMMA manual:

"Specifically,  if  SNPs  with  lower  minor  allele  frequency  tend  to
have larger effects (which is inversely proportional to its genotype variance), then the standardized
genotype matrix is preferred.  If the SNP effect size does not depend on its minor allele frequency,
then  the  centered  genotype  matrix  is  preferred.   In  our  previous  experience  based  on  a  limited
examples,  we  typically  find  the  centered  genotype  matrix  provides  better  control  for  population
structure in lower organisms, and the two matrices seem to perform similarly in humans."

I used the standardized calculation:

$$G_s = \frac{1}{p}\sum^p_{i=1}\frac{1}{v_{x_i}} (x_i - 1_n \bar x_i)(x_i - 1_n \bar x_i)^T$$

Th standardized calculation of relatedness takes into account the sample's variance of the $i$th SNP, whereas the centered calculation does not have the $\frac{1}{v_{x_i}}$ information. As stated above, this is preferable
when analyzing data where rare SNPs tend to have larger effects. [I suppose that I can check this in my GWAS 
output, or is this more of a theoretical thing?]

Also, GEMMA filters for SNPs. Out of 408,043 SNPs in the bimbam input, GEMMA analyzed 53,479 SNPs. This is increased from previous versions of this
with the maize lines with high frequency of missing values from 48K SNPs.

Why such a large reduction in SNPs analyzed?

Correlation with covariate?: "By default, SNPs with $r^2$ correlation with any of the covariates
above 0.9999 will not be included in the analysis." But we didn't put in a covariate matrix, so it can't be this?

```{r}
relMat <- read.table("data/trait_gwas_output/GDD/output/ZeaGBSv27_sF_F.sXX.txt")

image(as.matrix(relMat))
```

Population Information on the 490 lines is in progress.

#GEMMA Univariate Linear Mixed Model

The equation:

$$y = W \alpha + x \beta + u + \epsilon$$
(but without $W\alpha$?)

$$u \sim MVN_n(0,\lambda\tau^{-1}K), \epsilon \sim MVN_n(0,\tau^{-1}I_n)$$

$y$ = n-vector of quantitative traits for $n$ individuals  
$W$ = nxc matrix of covariates (fixed effects)  
$\alpha$ = c-vector of corresponding coefficients including the intercept  
$x$ = n-vector of marker genotypes  
$\beta$ = effect size of the marker  
$u$ = n-vector of random effects  
$\epsilon$ = n-vector or errors  
$\tau^{-1}$ = variance of the residual errors  
$\lambda$ = ratio between the two variance components  
$K$ = known nxn relatedness matrix  
$I_n$ = nxn identity matrix  
$MVN_n$ = n-dimensional multivariate normal distribution

```{r, fig.width=12, fig.height=8}
for (i in 1:length(traits)) {
  t <- as.character(traits[1,i])
  
  # trait GWAS with maize lines with high freq missing
  #gwas_path <- paste(paste("data/lowerFreq/", t, sep=""), "/", sep="")
  
  #assoc_file <- paste(paste("output/ZeaGBSv27_", t, sep=""), ".assoc.txt", sep="")
  #print(paste(gwas_path, assoc_file, sep=""))
  #assoc <- read.table(paste(gwas_path, assoc_file, sep=""), sep="\t", header=T)
  #print(dim(assoc))
  
  # trait GWAS with the individiduals with high freq missing data removed
  gwas_rm <- paste(paste("data/trait_gwas_output/", t, sep=""), "/", sep="")
  rm_assoc_file <- paste(paste("output/ZeaGBSv27_", t, sep=""), ".assoc.txt", sep="")
  
  assoc2 <- read.table(paste(gwas_rm, rm_assoc_file, sep=""), sep="\t", header=T)
  
  #dev.off()
  #par(mfrow=c(2,2), las=1)
  #manhattan(assoc, chr="chr", snp="rs", bp="ps", p="p_score", main=t)
  #qq(assoc$p_score, main=paste(t, "Q-Q plot of GWAS p-values", sep="\n"))
  manhattan(assoc2, chr="chr", snp="rs", bp="ps", p="p_score", main=t)
  qq(assoc2$p_score, main=paste(t, "Q-Q plot of GWAS p-values", sep="\n"))
}
```

Things to do:

local false sign rate and ASHR

Matthew Stephens: https://academic.oup.com/biostatistics/article/doi/10.1093/biostatistics/kxw041/2557030/False-discovery-rates-a-new-deal 
(FDR, a new deal)

ashr: https://github.com/stephens999/ashr

- give beta and se from gemma output

permutations in phenotype in lmm? to calibrate p-values

genomic control


Qx - take top X hits for each trait  
  Talk to Emily and Markus

Expression gwas

Gwas -
  pop structure
  dropping individuals/ why are snps being dropped
  